app/assets/v2/js/grants/index.js

```diff
@@ -23,6 +23,36 @@ $(document).ready(() => {
 
 });
 
+function debounce(func, wait) {
+  let timeout;
+  let immediate;
+
+  return function() {
+    // reference the context and args for the setTimeout function
+    let context = this; let
+        args = arguments;
+    
+    // allow user-triggered debounce from the func's supplied argument
+    if (args[0].debounce !== undefined && args[0].debounce === true)
+      immediate = false;
+    else if (args[0].debounce === undefined || args[0].debounce === false)
+      immediate = true;
+
+    const callNow = immediate && !timeout;
+
+    clearTimeout(timeout);
+    timeout = setTimeout(function() {
+      timeout = null;
+      if (!immediate) {
+        // call the original function with apply
+        func.apply(context, args);
+      }
+    }, wait);
+
+    if (callNow)
+      func.apply(context, args);
+  };
+}
```

Comment:
We have debounce methods in the codebase already, consider moving one of those to `shared.js` rather than implementing again 👍